kafka-persistence/src/main/java/com/netflix/conductor/dao/kafka/index/consumer/KafkaConsumer.java

```diff
@@ -0,0 +1,137 @@
+
+        Properties consumerConfig = new Properties();
+        consumerConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, elasticSearchConfiguration.getProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, ConsumerConstants.DEFAULT_BOOTSTRAP_SERVERS_CONFIG));
+        consumerConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ConsumerConstants.STRING_DESERIALIZER);
+        consumerConfig.put(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG, requestTimeoutMs);
+        consumerConfig.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ConsumerConstants.STRING_DESERIALIZER);
+        consumerConfig.put("group.id",  "_group");
+        consumer = new org.apache.kafka.clients.consumer.KafkaConsumer(consumerConfig);
+
+        consumer.subscribe(Collections.singleton(this.topic));
+
+        scheduler = Executors.newSingleThreadExecutor();
+        try {
+            scheduler.submit(() -> consume());
+        } catch(RejectedExecutionException e) {
+            logger.error("Task Rejected in scheduler Exception {}", e);
+        }
+    }
+
+    public void consume() {
+        while(true) {
+            try {
+                ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(this.pollInterval));
+                records.forEach(record -> consumeData(record.value()));
+            } catch (KafkaException e) {
+                logger.error("kafka KafkaConsumer message polling failed.", e);
+            }
+        }
+    }
+
+    public void consumeData(String data) {
+        try {
+            Record d = om.readValue(data, Record.class);
+            byte[] payload;
+            long start = System.currentTimeMillis();
+            switch (d.getType()) {
+                case RecordTypeConstants.WORKFLOW_DOC_TYPE:
+                    payload = om.writeValueAsBytes(d.getPayload());
+                    consumeWorkflow(payload, om.readTree(data).get("workflowId").asText());
+                    break;
+                case RecordTypeConstants.TASK_DOC_TYPE:
+                    payload = om.writeValueAsBytes(d.getPayload());
+                    consumeTask(payload, om.readTree(data).get("taskId").asText());
+                    break;
+                default:
+                    break;
+            }
+            Monitors.getTimer(Monitors.classQualifier, "elastic_search_index_time", "elastic_search_index_time").record(System.currentTimeMillis() - start, TimeUnit.MILLISECONDS);
+        } catch (IOException e) {
+            // JSON is not formatted. Workflow details in UI won't be available.
+            logger.error("Failed to consume from kafka - unknown exception:", e);
+        }
+    }
+
+    private void consumeWorkflow(byte[] doc, String id) {
+        try {
+            WorkflowSummary workflowSummary  = om.readValue(new String(doc), WorkflowSummary.class);
+            indexDAO.asyncIndexWorkflowSummary(workflowSummary);
```

Comment:
@apanicker-nflx ,The same indexDAO is injected in KafkaConsumer. So I can't call indexWorkflow here because it will again publish to kafka which should not happen. Here the actual indexing call must be made.

Reply:
This is the reason I have asked to break IndexDAO in two parts produceDAO and consumeDAO.